custom_components/ttlock/services.py

```diff
@@ -86,6 +88,19 @@ def register(self) -> None:
             supports_response=SupportsResponse.OPTIONAL,
         )
 
+        self.hass.services.register(
+            DOMAIN,
+            SVC_SET_AUTOLOCK,
+            self.handle_set_autolock,
+            vol.Schema(
+                {
+                    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+                    vol.Required(CONF_ENABLED): cv.boolean,
+                    vol.Optional(CONF_SECONDS): cv.positive_int,
```

Comment:
You can be more specific here if it's useful, something like this I think
```suggestion
                    vol.Optional(CONF_SECONDS): vol.All(vol.Coerce(int), vol.Range(min=0, max=60)),
```

Reply:
Yes I hacked this is the **_services.yaml_** file as I have no idea what vol or cv actually are at all 😂
Would that mean the **_services.yaml_** file wouldn't need the selector or do the 2 files have nothing to do with each other (i looked at it like the .py file was actual backend where the .yaml was just for appearances)

Reply:
`cv` stands for config validation and is Home Assistants wrapper around a library.
That library `vol` or `voluptuous` is the underlying thing that does the validations.

The suggested line is essentially putting together their building blocks by saying:
- match [all](http://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#voluptuous.validators.All) the conditions
- must [be an](http://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#voluptuous.validators.Coerce) int
- in the [range](http://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#voluptuous.validators.Range) of 0 to 60